src/components/textField/TextField.tsx

```diff
@@ -0,0 +1,151 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+import { Text, TextInput, TouchableWithoutFeedback, View } from 'react-native';
+import ThemeContext from '@contexts/ThemeContext';
+import Icon from '@components/icon';
+import IconButton from '@components/iconButton';
+import textfieldStyles from './styles';
+
+const TextField = ({
+  value,
+  onChangeText,
+  placeholder,
+  label,
+  ref,
+  autoFocus,
+  keyboardType,
+  inputMode,
+  enterKeyHint,
+  onSubmitEditing,
+  onEndEditing,
+  errorMsg,
+  isPassword,
+  prefixIcon,
+  suffixIcon,
+  suffixIconButton,
+  containerStyle,
+  textInputStyle,
+  addOns,
+}: TextFieldProps) => {
+  const {
+    theme: { textfield: theme },
+  } = useContext(ThemeContext);
+
+  const inputRef = useRef<TextInput>(null);
+
+  const [styles] = useState(textfieldStyles());
```

Comment:
why using use state here?'

Reply:
Yes
because i used context in styles 
so if i do without state every time when component re renders it again calls the context of styles so the react is giving warning on calling multiple context

it is not giving this warning in this component
but in some other screen and component it is giving warning

so i used this method

Reply:
i know this is not a good practice 
and i will be upgrading the code for managing theme ahead in this project

Reply:
make sure to update this